lua/oil/adapters/files.lua

```diff
@@ -213,12 +213,28 @@ end
 M.normalize_url = function(url, callback)
   local scheme, path = util.parse_url(url)
   assert(path)
-  if fs.is_windows and path == "/" then
-    return callback(url)
+
+  if fs.is_windows then
+    if path == "/" then
+      return callback(url)
+    else
+      local drive = url:match("^oil:///(%u)$")
+      if drive then
+        return callback("oil:///" .. drive .. "/")
+      end
+    end
   end
+
   local os_path = vim.fn.fnamemodify(fs.posix_to_os_path(path), ":p")
   uv.fs_realpath(os_path, function(err, new_os_path)
-    local realpath = new_os_path or os_path
+    local realpath
+
+    if fs.is_windows then
+      realpath = os_path or new_os_path
+    else
+      realpath = new_os_path or new_os_path
+    end
```

Comment:
This section handles fixed mapped network drives, as it was resolving to the UNC path, in my case the \\ IP address rather than the drive letter that was mapped

Reply:
Then if `fs.is_windows` is true we expect `realpath` will only ever be set to `os_path`, because `os_path` is not nil. Can we do this as an early return instead and avoid calling `fs_realpath` entirely?